File: StriverDP/uniquePaths.py
'''
https://www.codingninjas.com/codestudio/problems/total-unique-paths_1081470?leftPanelTab=3
'''
from os import *
from sys import *
from collections import *
from math import *

def uniquePaths(m, n):
	m,n = int(m),int(n)
	dp = [1] * m
	for i in range(1, n):
		for j in range(1, m):
			dp[j] += dp[j-1]
	return dp[-1]

File: StriverDP/UniquePaths.cpp
/*
https://www.codingninjas.com/codestudio/problems/total-unique-paths_1081470?leftPanelTab=0
*/
#include <bits/stdc++.h> 
int uniquePaths(int m, int n) {
	int N = n+m-2;
	int R = n-1;
	double result = 1.0;
	for(int i = 1 ; i <= R ; i++){
		result = result*(N-R+i)/i;
	}
	return (int)result;
}
